### PR TITLE
fix: concurrent map iteration and map write

### LINKED ISSUE #726: fatal error: concurrent map iteration and map write
### Report

High concurrent long-term requesting, the interceptor component will restart after a period of time. `kubectl logs podxxx` the pod log show `fatal error: concurrent map iteration and map write`
image

### Expected Behavior

`interceptor` should be stable and don't restart when high concurrent requesting.

### Actual Behavior

`interceptor` component restarts many time.

### Steps to Reproduce the Problem

1. All the components is runing.
2. Using jmeter to send many request to `nginx`

### Logs from KEDA HTTP operator

```
example
```

### HTTP Add-on Version

0.4.0

### Kubernetes Version

1.26

### Platform

Any

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by luckymrwang**
_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Any necessary documentation is added, such as:
  - [`README.md`](/README.md)
  - [The `docs/` directory](./docs)
  - [The docs repo](https://github.com/kedacore/keda-docs)

Fixes #726

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
Thanks for the fix! ❤️ 
Could you update changelog as well?

File: pkg/queue/queue.go [hunk 1]
```diff
@@ -85,6 +85,8 @@ func (r *Memory) Current() (*Counts, error) {
 	r.mut.RLock()
 	defer r.mut.RUnlock()
 	cts := NewCounts()
-	cts.Counts = r.countMap
+	for key, val := range r.countMap {
+		cts.Counts[key] = val
+	}
```
[the remark this answers is not in the document]
Comment (t0rr3sp3dr0) [reply pkg/queue/queue.go:90]: This should be fine. The problem is that we were passing the reference of the internal Queue map, that is protected by a mutex, to the Counts struct, that is not aware of the mutex. So when someone interacted with the Counts struct, the internal Queue map was being used without locking the mutex. Now the map is being copied to the Counts struct, fixing the problem.

[the remark this answers is not in the document]
Comment (t0rr3sp3dr0) [reply pkg/queue/queue.go:90]: The Counts struct is not thread-safe and its instances are not being accessed by multiple routines, so we don’t need a mutex that side.

**Review by t0rr3sp3dr0 — COMMENTED** [review]
@luckymrwang, to prevent regressions in the future, can you implement a test to make sure the Current method always returns a copy of the internal map and not a reference to it? Given we are now returning a new map, it’s a good idea to also test if the Counts map contains the same values as the Queue map.

**Comment by luckymrwang** [comment]
> @luckymrwang, to prevent regressions in the future, can you implement a test to make sure the Current method always returns a copy of the internal map and not a reference to it? Given we are now returning a new map, it’s a good idea to also test if the Counts map contains the same values as the Queue map.

@t0rr3sp3dr0 Ok, I will add it later.

**Comment by JorTurFer** [comment]
Hi @luckymrwang ,
We plan to cut a hotfix release soon (this week). Do you think that you will have enough time?

**Comment by luckymrwang** [comment]
> Hi @luckymrwang , We plan to cut a hotfix release soon (this week). Do you think that you will have enough time?

@JorTurFer @t0rr3sp3dr0  I was busy the past two days and the latest code has been submitted.

**Comment by JorTurFer** [comment]
Could you update changelog as well?

**Comment by luckymrwang** [comment]
> Statics checks are faillig, could you review them and update changelog as well?

@JorTurFer The latest code has been committed.

**Review by JorTurFer — COMMENTED** [review]
Please, add a note in the changelog about this PR: https://github.com/kedacore/http-add-on/blob/main/CHANGELOG.md#fixes

**Review by t0rr3sp3dr0 — APPROVED** [review]

**Comment by t0rr3sp3dr0** [comment]
lgtm

**Pull request merged**